src/buffer/out/textBuffer.cpp

```diff
@@ -1440,10 +1440,23 @@ til::point TextBuffer::_GetWordStartForSelection(const til::point target, const
     // expand left until we hit the left boundary or a different delimiter class
     while (result != bufferSize.Origin() && _GetDelimiterClassAt(result, wordDelimiters) == initialDelimiter)
     {
-        //prevent selection wrapping on whitespace selection
-        if (isControlChar && result.x == bufferSize.Left())
+        if (result.x == bufferSize.Left())
         {
-            break;
+            // Prevent wrapping to the previous line if the selection begins on whitespace
+            if (isControlChar)
+            {
+                break;
+            }

```

Comment:
...but should we? If I print "a    b" and it force-wraps in the middle, shouldn't I be able to select all 4 spaces with a double click?

Reply:
Eh, I didn't want to revisit too many of the decisions from the previous code.

But also, this is what prevents us from selecting **the entire text buffer** when you double-click an empty space lmao